js/lib.globals.d.ts

```diff
@@ -0,0 +1,19 @@
+// Copyright 2018 the Deno authors. All rights reserved. MIT license.
+
+// This file contains the default TypeScript libraries for the runtime
+
+/// <reference no-default-lib="true"/>
+
+/// <reference lib="esnext" />
+
+import "gen/js/globals";
+
+interface Window {
+  // TODO(ry) These shouldn't be global.
+  mainSource: string;
+  setMainSourceMap(sm: string): void;
+}
+
+// TODO(ry) These shouldn't be global.
+declare let mainSource: string;
+declare function setMainSourceMap(sm: string): void;
```

Comment:
Should this file be removed when `mainSource` and `setMainSourceMap` are moved? If so can you add a TODO?

Reply:
I wasn't totally sure, as in the future it is hard to predict where all the runtime definitions will come from... right now it is `globals.ts`.  On the other hand, if we focus on `globals.ts` doing all of the runtime definition of what is exposed in the global scope (or pulling in other modules) we can use its generated definition as the default lib returned from the compiler.  I will put a todo in there to revisit though.

Reply:
Yeah - let's see how things look after the TextDecoder/TextEncoder (#452) lands. I think that will clarify things.